docs/Questions.md

```diff
@@ -0,0 +1,98 @@
+<H1> Folder structure(Pages, Components, Templates, Organisms, Molecules, Atoms) </H1>
+We should stick to one way of organizing our folder and component's structure.
+Utils have files with .tsx file extension, which I think is inappropriate.
+We have validation.js and wdyr.js files at the top level of src
+
+<H1> Styles </H1>
+Currently we both use Styled-Components and scss across our codebase. I think we should stick to one and remove the other
+
+---
+
+Mike's comment: _Personally, I prefer styled-components because they allow you to pass variables to your styles from the component_
+
+<H1> Data fetching & selectors in components </H1>
+We have a lot of ambiguity about how we interact with data in our app.
+For example:
+
+- `useConnects` in components with different ways of solving the issue, when there is no needed data(like a venue id)
+- firestore direct pushes in `actions` and components
+
+<H1> Merging flow </H1>
+We should have a well-defined merging flow to avoid misunderstandings.
+
+---
+
+Mike's comment:
+
+Depending on the CI\CD strategy we choose, we could end up with 2 flows(normal | hotfix).
+
+- Normal: feat|impr|fix branch => Staging => Production
+- Hotfix: hotfix branch => Prod && hotfix branch => Staging
+
+<H1> Branch and commit naming </H1>
+Well-defined rules for this can help keep everything tidy and understandable by all parties
```

Comment:
I suggest having only 4 allowed terms in commits, branches, PR names:

- Improvement | IMPR - Refactoring
- Fix | FIX - bug fix
- Hotfix | HOTFIX - Direct bugfix to production(hopefully, the rarest one)
- Feature | FEAT - New feature / Behavior change of an old feature

Example:

- _[GITHUB_BOARD_TICKET_NUMBER]/['feat' | 'fix' | 'impr' | 'hotfix']/[short branch description] as an example of branch naming. Should be solidified as a regexp rule_
- _[GITHUB_BOARD_TICKET_NUMBER]: [Commit extensive description in present simple] - commit naming example. Should be solidified as a regexp rule_

Reply:
I agree in general, but as mentioned above, I tend to strongly preference existing standards.

As far as commit naming/etc goes, there are some very solid standards that I have a strong preference towards. I haven't figured out the nuanced detail within them yet, as there are a couple of options, but at a very high level, take a look at things like (pulled from some of my personal notes on this sort of topic):

- https://www.conventionalcommits.org/
- https://github.com/conventional-changelog/conventional-changelog
- https://github.com/conventional-changelog/standard-version
- https://commitlint.js.org/
- http://commitizen.github.io/cz-cli/
- https://github.com/commitizen/cz-conventional-changelog
- https://gitmoji.carloscuesta.me/
- https://semver.org/
- etc

Reply:
See also: https://github.com/sparkletown/internal-sparkle-docs/pull/1#discussion_r579889314